chirp-impl/src/test/java/sample/chirper/chirp/impl/ChirpServiceTest.java

```diff
@@ -17,17 +19,36 @@
 import scala.concurrent.duration.FiniteDuration;
 
 import java.time.Instant;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
 
 import static com.lightbend.lagom.javadsl.testkit.ServiceTest.*;
 import static java.util.concurrent.TimeUnit.SECONDS;
 
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
 public class ChirpServiceTest {
 
     private static TestServer server;
 
+    // Creating a new persistent entity is slow in some runtime environment (e.g. free tier Travis) causing the
+    // tests to fail on timeout. First time we invoke a call that requires a persistent entity we will
+    // wait a bit longer.
+    private static int firstRequestOnEntityTimeout;
+    private static final int secondRequestOnEntityTimeout = 1;
+
     @BeforeClass
-    public static void setUp() {
+    public static void setUp() throws InterruptedException, ExecutionException, TimeoutException {
         server = startServer(defaultSetup().withCassandra(true));
+
+        // This is a canary wait to ensure the server is up and running so tests can start. If
+        // this fails it's not a test failure, it's that the machine running this is slow.
+        ChirpService chirpService = server.client(ChirpService.class);
+        LiveChirpsRequest request = new LiveChirpsRequest(TreePVector.<String>empty().plus("usr1").plus("usr2"));
+        chirpService.getLiveChirps().invoke(request).toCompletableFuture().get(10, SECONDS);
+
+        // First time a given PE is requested it may take longer to respond. Let's run the tests with
+        // timeouts equivalent to production settings _plus_ one second.
+        firstRequestOnEntityTimeout = 1 + (int) server.app().config().getDuration("lagom.persistence.ask-timeout", SECONDS);
```

Comment:
This configures the timeouts so that the request timeout (from the test PoV) is `1 + ask-timeout`, where `ask-timeout` is the configured timeout when invoking `ask()` on a `PersistentEntityRef`.

The intention is to make the test fail in production code not on test code.